tests/mgbench/dataset_creator.py

```diff
@@ -0,0 +1,120 @@
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import random
+import helpers
+import argparse
+
+# Explaination of datasets:
+#   - empty_only_index: contains index; contains no data
+#   - small: contains index; contains data (small dataset)
+#
+# Datamodel is as follow:
+#
+#                               ┌──────────────┐
+#                               │ Permission   │
+#         ┌────────────────┐    │  Schema:uuid │   ┌────────────┐
+#         │:IS_FOR_IDENTITY├────┤  Index:name  ├───┤:IS_FOR_FILE│
+#         └┬───────────────┘    └──────────────┘   └────────────┤
+#          │                                                    │
+#   ┌──────▼──────────────┐                                  ┌──▼────────────────┐
+#   │  Identity           │                                  │ File              │
+#   │   Schema:uuid       │                                  │  Schema:uuid      │
+#   │   Index:email       │                                  │  Index:name       │
+#   └─────────────────────┘                                  │  Index:platformId │
+#                                                            └───────────────────┘
+#
+#   - File: attributes: ["uuid", "name", "platformId"]
+#   - Permission: attributes: ["uuid", "name"]
+#   - Identity: attributes: ["uuid", "email"]
+#
+# Indexes:
+#   - File: [File(uuid), File(platformId), File(name)]
+#   - Permission: [Permission(uuid), Permission(name)]
+#   - Identity: [Identity(uuid), Identity(email)]
+#
+# Edges:
+#   - (:Permission)-[:IS_FOR_FILE]->(:File)
+#   - (:Permission)-[:IS_FOR_IDENTITYR]->(:Identity)
+#
+# AccessControl specific: uuid is the schema
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--number_of_identities", type=int, default=10)
+    parser.add_argument("--number_of_files", type=int, default=10)
+    parser.add_argument("--percentage_of_permissions", type=float, default=1.0)
+    parser.add_argument("--filename", default="dataset.cypher")
+
+    args = parser.parse_args()
+
+    number_of_identities = args.number_of_identities
+    number_of_files = args.number_of_files
+    percentage_of_permissions = args.percentage_of_permissions
+    filename = args.filename
+
+    assert number_of_identities >= 0
+    assert number_of_files >= 0
+    assert percentage_of_permissions > 0.0 and percentage_of_permissions <= 1.0
+    assert filename != ""
+
+    f = open(filename, "w")
+
+    f.write("MATCH (n) DETACH DELETE n;\n")
+
+    # Create the indexes
+    f.write("CREATE INDEX ON :File;\n")
+    f.write("CREATE INDEX ON :Permission;\n")
+    f.write("CREATE INDEX ON :Identity;\n")
+    f.write("CREATE INDEX ON :File(platformId);\n")
+    f.write("CREATE INDEX ON :File(name);\n")
+    f.write("CREATE INDEX ON :Permission(name);\n")
+    f.write("CREATE INDEX ON :Identity(email);\n")
```

Comment:
Maybe use with, not sure how it will look after so many indentations:
```suggestion
    with open(filename, "w") as f:

      f.write("MATCH (n) DETACH DELETE n;\n")

      # Create the indexes
      f.write("CREATE INDEX ON :File;\n")
      f.write("CREATE INDEX ON :Permission;\n")
      f.write("CREATE INDEX ON :Identity;\n")
      f.write("CREATE INDEX ON :File(platformId);\n")
      f.write("CREATE INDEX ON :File(name);\n")
      f.write("CREATE INDEX ON :Permission(name);\n")
      f.write("CREATE INDEX ON :Identity(email);\n")
```
and you can remove close